extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -482,8 +529,14 @@ export class AmpA4A extends AMP.BaseElement {
                   if (isValid) {
                     return creative;
                   }
-                  return Promise.reject(
-                      'Key failed to validate creative\'s signature.');
+                  // Only report is signature is expected to match given
+                  // multiple key providers could have been specified.
+                  if (!verifyHashVersion(signature, keyInfo)) {
```

Comment:
Can you explain why this won't produce false positives? If you've got a locally built AMP runtime, this will run against both the dev and prod keys, and if the creative was signed with the prod key `verifyHashVersion` will return true for the dev key. Unless I've misunderstood something?